enf/network.go

```diff
@@ -6,31 +6,41 @@ import (
 	"net/http"
 )
 
+// NetworkService handles communication with the network related
+// methods of the ENF API. These methods are used to manage the networks
+// under each domain.
 type NetworkService service
 
+// NetworkRequest is used to create a new network in Network.CreateNetwork.
 type NetworkRequest struct {
 	Name        *string `json:"name"`
 	Description *string `json:"description"`
 }
 
+// Network represents a network in the ENF.
 type Network struct {
 	Name        *string `json:"name"`
 	Network     *string `json:"network"`
 	Description *string `json:"description"`
 	Status      *string `json:"status"`
 }
 
+// NetworkResponse represents the typical API response for all
+// the endpoints related to the Network resource.
+type NetworkResponse struct {
```

Comment:
The "response" structs were originally anonymous, because I don't want to expose the "data" member, unused "pages", etc. to `go-enf` clients.   Repeating the definition in each method is 

All of the `xcr` endpoints (not just `xcr/v2/networks`) share that same structure --- repeating that definition in each method is annoying, but I couldn't find a way in `go` to define it in a generic way.  The `Data[] *X` member needs a concrete type for `X`.

Defining it once per resource type, as you're doing here for `Network` seems like a reasonable middle ground.

Let's make it private though (`networkresponse`), so that users of `go-enf` don't have access to it.  In `go`, identifiers starting with a lower case letter are private; those starting with an uppercase letter are public.